cpp/tests/utilities/column_utilities.cu

```diff
@@ -34,26 +39,97 @@ void expect_column_properties_equal(cudf::column_view lhs, cudf::column_view rhs
-      EXPECT_EQ(lhs.nullable(), rhs.nullable());
+     EXPECT_EQ(lhs.nullable(), rhs.nullable());
   }
   EXPECT_EQ(lhs.has_nulls(), rhs.has_nulls());
   EXPECT_EQ(lhs.num_children(), rhs.num_children());
 }
 
-// Verify elementwise equality
-void expect_columns_equal(cudf::column_view lhs, cudf::column_view rhs) {
+class corresponding_rows_unequal {
+public:
+  corresponding_rows_unequal(table_device_view d_lhs, table_device_view d_rhs): comp(d_lhs, d_rhs) {
+  }
+  
+  cudf::experimental::row_equality_comparator<true> comp;
+    
+  __device__ bool operator()(size_type index) {
+    return !comp(index, index);
+  }
+};
+
+void expect_columns_equal(cudf::column_view lhs, cudf::column_view rhs, bool print_all_differences) {
   expect_column_properties_equal(lhs, rhs);
 
   auto d_lhs = cudf::table_device_view::create(table_view{{lhs}});
   auto d_rhs = cudf::table_device_view::create(table_view{{rhs}});
 
-  EXPECT_TRUE(
-      thrust::equal(thrust::device, thrust::make_counting_iterator(0),
-                    thrust::make_counting_iterator(lhs.size()),
-                    thrust::make_counting_iterator(0),
-                    cudf::experimental::row_equality_comparator<true>{*d_lhs, *d_rhs}));
+  thrust::device_vector<int> differences;
+  differences.reserve(lhs.size());
+
+  auto diff_iter = thrust::copy_if(thrust::device,
+                                   thrust::make_counting_iterator(0),
+                                   thrust::make_counting_iterator(lhs.size()),
+                                   differences.begin(),
+                                   corresponding_rows_unequal(*d_lhs, *d_rhs));
 
   CUDA_TRY(cudaDeviceSynchronize());
+
+  if (diff_iter > differences.begin()) {
+    int difference_count = diff_iter - differences.begin();
+
+    if (print_all_differences) {
+      //
+      //  If there are differences, display them all
+      //
+      std::ostringstream buffer;
+      buffer << "differences:" << std::endl;
+      
+      cudf::table_view source_table ({lhs, rhs});
+
+      fixed_width_column_wrapper<int> diff_column(differences.begin(), diff_iter);
+
+      std::unique_ptr<cudf::experimental::table> diff_table = cudf::experimental::gather(source_table,
+											 diff_column);
+      
+      //
+      //  Need to pull back the differences
+      //
+      std::vector<std::string> h_left_strings = to_strings(diff_table->get_column(0));
+      std::vector<std::string> h_right_strings = to_strings(diff_table->get_column(1));
+
+      for (int i = 0 ; i < difference_count ; ++i) {
+          buffer << "lhs[" << differences[i] << "] = " << h_left_strings[i]
+                 << ", rhs[" << differences[i] << "] = " << h_right_strings[i] << std::endl;
+      }
+
+      EXPECT_EQ(difference_count, 0) << buffer.str();
+    } else {
+      //
+      //  If there are differences, just display the first one
+      //
+      int index = differences[0];
+
+      cudf::column_view diff_lhs(lhs.type(),
```

Comment:
This wont work for a `strings` column. 

I think the easiest way to do it and make it work for a strings column is to slice off just the first element using https://github.com/rapidsai/cudf/blob/c51184301b041894aa55778509b9d67d4fd644e5/cpp/include/cudf/detail/copy.hpp#L44 and then do `to_string` on the slice.

Reply:
Created a test with strings column forcing a failure.  The slice you suggested works.

The print_all_differences version doesn't work for strings at the moment, since gather throws an exception if you specify a string column.  Although it looks like it would.

Reply:
Just pushed an update that reflects these changes.

Reply:
> The print_all_differences version doesn't work for strings at the moment, since gather throws an exception if you specify a string column. Although it looks like it would.

That's expected, gather doesn't support strings yet, but it will soon. I don't think we need to hold up your PR waiting for that to happen.